src/otx/core/model/entity/visual_prompting.py

```diff
@@ -0,0 +1,26 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Class definition for visual prompting model entity used in OTX."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from otx.core.data.entity.visual_prompting import VisualPromptingBatchDataEntity, VisualPromptingBatchPredEntity
+from otx.core.model.entity.base import OTXModel
+from otx.core.utils.config import inplace_num_classes
+
+if TYPE_CHECKING:
+    from omegaconf import DictConfig
+
+
+class OTXVisualPromptingModel(
+    OTXModel[VisualPromptingBatchDataEntity, VisualPromptingBatchPredEntity],
+):
+    """Base class for the visual prompting models used in OTX."""
+
+    def __init__(self, config: DictConfig, num_classes: int) -> None:
```

Comment:
```suggestion
    def __init__(self, num_classes: int) -> None:
```

Do we need `config` here? Is your model based on MM*? In addition, is `num_classes` meaningful in this task?

Reply:
I enveloped model configs into `config` like other model entities to use them easily, but if you think that getting parameters respectively would be better, I'll update it.
And as you said `num_classes` is not used for training this task, but base otx model uses `num_classes` as default, I had no choice but to set dummy `num_classes` in recipe.

https://github.com/openvinotoolkit/training_extensions/blob/c1cec1a336e05b82260181d955a2ce4e89be6570/src/otx/core/model/entity/base.py#L32-L44

Reply:
If so, please remove `num_classes` from its argument such as
```python
     def __init__(self) -> None:
         super().__init__(num_classes=0)
```
We've used `config` since introducing it is comfortable for integrating MM* models, but this task doesn't require MM*. I think you can hide `config` into `OTXSegmentAnything.__init__()` such as

```python
class OTXSegmentAnything:
    def __init__(self, variant: Literal["mobilesam", ...]):
        if variant == "mobilesam":
              self._create_model(mobilesam_config) # mobilesam_config is defined somewhere
```

Reply:
Thanks, and I have one more question.
`model/default.yaml` has `num_classes` so I cannot ignore in the recipe.

https://github.com/openvinotoolkit/training_extensions/blob/c1cec1a336e05b82260181d955a2ce4e89be6570/src/otx/config/model/default.yaml#L14-L15

Reply:
Hmm Sorry I didn't catch it then can we 
```python
     def __init__(self, num_classes: int = 0) -> None:
         super().__init__(num_classes=num_classes)
```

Reply:
Fixed at https://github.com/openvinotoolkit/training_extensions/pull/2780/commits/327051b0a57283f21c2ccaac362730418c5516cc